impeller/entity/contents/filters/blend_filter_contents.cc

```diff
@@ -178,6 +179,150 @@ static std::optional<Entity> AdvancedBlend(
       entity.GetBlendMode(), entity.GetStencilDepth());
 }
 
+Entity BlendFilterContents::CreateForgroundBlend(
+    const std::shared_ptr<FilterInput>& input,
+    const ContentContext& renderer,
+    const Entity& entity,
+    const Rect& coverage,
+    Color foreground_color,
+    BlendMode blend_mode) const {
+  RenderProc render_proc = [foreground_color, coverage, input, blend_mode](
+                               const ContentContext& renderer,
+                               const Entity& entity, RenderPass& pass) -> bool {
+    using VS = BlendScreenPipeline::VertexShader;
+    using FS = BlendScreenPipeline::FragmentShader;
+
+    auto& host_buffer = pass.GetTransientsBuffer();
+
+    auto dst_snapshot = input->GetSnapshot(renderer, entity);
+    if (!dst_snapshot.has_value()) {
+      return false;
+    }
+
+    // This does not use the snapshot transform, since this value will
+    // be different depending on whether the filter was applied via a
+    // saveLayer or directly to a particular contents. By construction,
+    // the provided rectangle should exactly map to the texture coordinates.
```

Comment:
Got it, I took another look at this and found that I was more or less applying the origin offset twice since I was still appending the transform shift to the entity transform - which explained the difference.

This now uses getSnapshotUVs and renders correctly in both scenarios.